pkg/ioutil/write_closer_test.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ioutil
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWriteCloseInformer(t *testing.T) {
+	original := &writeCloser{}
+	wci, close := NewWriteCloseInformer(original)
+	data := "test"
+
+	n, err := wci.Write([]byte(data))
+	assert.Equal(t, len(data), n)
+	assert.Equal(t, data, original.buf.String())
+	assert.NoError(t, err)
+
+	select {
+	case <-close:
+		assert.Fail(t, "write closer closed")
+	default:
+	}
+
+	wci.Close()
+	assert.True(t, original.closed)
+
+	select {
+	case <-close:
+	default:
+		assert.Fail(t, "write closer not closed")
+	}
+}
```

Comment:
We mainly want to test `WriteCloseInformer` will close the inform channel when it is closed. And the test proves that it does. :)